### PR TITLE
Validate client request

### PR DESCRIPTION
**Pull request by stevenjack**
### Problem

We need to validate the PayPal IPN response based on the following spec https://developer.paypal.com/docs/classic/ipn/integration-guide/IPNImplementation/#specs.

### Solution

* Create `Validate` method that takes a body, environment and http client and performs the steps necessary to validate the response.

### Notes

- **Remember** to tag your merge commit after merging 👍

### CODE REVIEW

**Comment by stevenjack** [comment]
[[GPI 0.3.0] Send back response to validate request.](https://trello.com/c/Yx0nrSp8/937-gpi-0-3-0-send-back-response-to-validate-request)

File: validator.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		return false, err
+	}
+
+	switch validationBody {
+	case "VERIFIED":
+		return true, nil
+	case "INVALID":
+		return false, errors.New("IPN payload sent by PayPal didn't match payload received")
+	default:
+		return false, fmt.Errorf("Expected response to include: 'VERIFIED' or 'INVALID' but got: '%s'", validationBody)
```
Comment (revett) [inline validator.go]: Multi-line!

File: validator.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	validationBody, err := ReadBodyIntoString(response.Body)
+	if err != nil {
+		return false, err
+	}
+
+	switch validationBody {
+	case "VERIFIED":
+		return true, nil
+	case "INVALID":
+		return false, errors.New("IPN payload sent by PayPal didn't match payload received")
```
Comment (revett) [inline validator.go]: Multi-line!

**Review by revett — APPROVED** [review]
Watch out for the 80 character line limit!

Nice size PR this time 👍

**Pull request merged**